### ISSUE TITLE
Opening a window with pyglet backend failed on Win10-10.0.18362

### ISSUE THREAD

**Issue by DingkunLiu** [psychopy/psychopy#2621]
The error comes when I open a window with pyglet backend. Here comes the error message:
```Hello from the pygame community. https://www.pygame.org/contribute.html
Paths to files on the system:
    userPrefsFile: C:\Users\liudk\AppData\Roaming\psychopy3\userPrefs.cfg
    appDataFile: C:\Users\liudk\AppData\Roaming\psychopy3\appData.cfg
    demos: C:\Users\liudk\.conda\envs\py36\lib\site-packages\psychopy\demos
    appFile: C:\Users\liudk\.conda\envs\py36\lib\site-packages\psychopy\app\PsychoPy.py

System info:
Windows-10-10.0.18362-SP0

Python info
C:\Users\liudk\.conda\envs\py36\python.exe
3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 14:00:49) [MSC v.1915 64 bit (AMD64)]
numpy 1.17.0
scipy 1.3.1
matplotlib 3.0.3
pyglet 1.3.2
pyo [not installed]

PsychoPy 3.1.5
Traceback (most recent call last):
  File ".\sysInfo.py", line 41, in <module>
    win = visual.Window([100, 100])  # some drivers want a window open first
  File "C:\Users\liudk\.conda\envs\py36\lib\site-packages\psychopy\visual\window.py", line 389, in __init__
    self.backend = backends.getBackend(win=self, *args, **kwargs)
  File "C:\Users\liudk\.conda\envs\py36\lib\site-packages\psychopy\visual\backends\__init__.py", line 32, in getBackend
    return Backend(win, *args, **kwargs)
  File "C:\Users\liudk\.conda\envs\py36\lib\site-packages\psychopy\visual\backends\pygletbackend.py", line 227, in __init__
    self._origGammaRamp = self.getGammaRamp()
  File "C:\Users\liudk\.conda\envs\py36\lib\site-packages\psychopy\visual\backends\pygletbackend.py", line 329, in getGammaRamp
    return getGammaRamp(self.screenID, self.xDisplay)
  File "C:\Users\liudk\.conda\envs\py36\lib\site-packages\psychopy\visual\backends\gamma.py", line 128, in getGammaRamp
    raise AssertionError('GetDeviceGammaRamp failed')
AssertionError: GetDeviceGammaRamp failed
1.2735  WARNING         Monitor specification not found. Creating a temporary one...
Exception ignored in: <bound method Window.__del__ of <psychopy.visual.window.Window object at 0x000001CF8FEA6668>>
Traceback (most recent call last):
  File "C:\Users\liudk\.conda\envs\py36\lib\site-packages\psychopy\visual\window.py", line 501, in __del__
    self.close()
  File "C:\Users\liudk\.conda\envs\py36\lib\site-packages\psychopy\visual\window.py", line 1327, in close
    self.backend.close()  # moved here, dereferencing the window prevents
AttributeError: 'NoneType' object has no attribute 'close'
```

**Comment by DingkunLiu** [comment]
There is an existing discussion on this issue. [link](https://discourse.psychopy.org/t/error-message-since-windows-update-to-1903/8355/2). The error seems to be caused by windows10 update. Hope it could be fixed soon. Thank you!

**Comment by peircej** [comment]
OK, I've got a new workaround for this to mitigate the issue (ultimately it's still Microsoft's issue).

There are various parts to the workaround:
1. for a lot of people they're having this when they don't even need gamma correction so the first step is to get rid of all attempts (and warnings or failures) for those users. We were already (correctly) checked the cases where gamma doesn't need setting at all, but in those cases we were still *getting* the original gamma ready to revert it on window lose. We don't need to. I've moved checking of the current gamma ramp to occur when (and only when) a new gamma/ramp is being set so that it doesn't get called when not needed.  
2. I've added [gamma problems info to the PsychoPy troubleshooting page](https://www.psychopy.org/troubleshooting.html#errors-with-getting-setting-the-gamma-ramp) for both the windows issue mentioned here, and also the Linux issue in #2061
3. The error/warning created by @djmannion now has an entry in the PsychoPy Prefences that can be set to apply for all experiemnts (Builder and code). That error also now points to the Troubleshooting link above.

These mitigations will be part of PsychoPy 3.2.4 (tomorrow?)

This doesn't fix the issue for users that need 
- Windows update 1093 installed (or just roll back)
- extended desktops (which appears to be the thing that clashes with Window 1903)
- gamma correction (many people don't)
